[DP-63] Add a GraphQL mutation to edit a word

chanmioh opened the pull request:
### CHANGES

- Defined a new InputObject to use with queries called AnnotationFormInput in `form.rs` that contains a word’s id and source_text 
- Added a new mutation in `query.rs` that takes in an AnnotationFormInput consisting of a word’s id and its new source_text field
- Added SQL call to update source text in `database_sql.rs` and added new SQL query called `update_source.sql` 
- Cleaned up `login.css.ts`, `login.page.tsx`, and `reset-password.page.tsx` styling a little bit

loafofpiecrust commented on graphql/src/query.rs:
```diff
@@ -288,6 +289,18 @@ impl Mutation {
         Ok(true)
     }
+
+    async fn update_source_text(
```
Hm I think the name `update_word` is better since we'll support more fields in the future. For new things I want to use the term `word` instead of `form` because it's just more obvious to people.

chanmioh replied:
That makes sense, I've went ahead and made the change!

loafofpiecrust commented on graphql/src/query.rs:
```diff
@@ -288,6 +289,18 @@ impl Mutation {
+    async fn update_source_text(
+        &self,
+        context: &Context<'_>,
+        annotation_form: AnnotationFormInput,
```
I would probably rename this argument to `word` and the type to `AnnotatedFormUpdate` or `UpdateAnnotatedForm` (just to keep consistent with the type it's based on).

loafofpiecrust commented on types/queries/update_word.sql:
```diff
@@ -0,0 +1 @@
+update word set source_text = $2 where id = $1
```
You'll need to run `dev-generate-types` since you made a SQL change.

chanmioh commented:
### CHANGES

- `query.rs`: Added guard to `update_word` function so only Editor group can access this resolver. Also just modified param names from `annotation_form` to `word` to keep it consistent
